### PR TITLE
Download kubeconfig for garden cluster

### LINKED ISSUE #417: Kubeconfig generation for user garden cluster access
Today, end-users can get their JWT IDToken in the Account page.
This is a hard process for several reasons 

- they have copy + paste that token in their `kubeconfig` with correct API server address, `ca-cert` and more.
- it's impossible to refresh it (no refresh token is provided).
- end-users have to login via the dashboard (which might not be okay for console users).

Ideally end-users can get a personalized `kubeconfig` when accessing the `Account` page:

```yaml
apiVersion: v1
clusters:
- cluster:
    server: https://MY-GARDEN-CLUSTER
  name: garden-oidc
contexts:
- context:
    cluster: garden-oidc
    user: garden-oidc
  name: garden-oidc
current-context: garden-oidc
kind: Config
preferences: {}
users:
- name: garden-oidc
  user:
    auth-provider:
      config:
        client-id: Test1234
        client-secret: Test1234
        id-token: SOME-ID-TOKEN
        idp-issuer-url: https://my-dex.foo.bar
        refresh-token: MY-REFRESH-TOKEN (applicable in some cases)
        extra-scopes: openid,email,profile,groups # coma separated scopes
      name: oidc
```

With this, end-users can access the garden cluster via `kubectl` and it'll automatically refresh the `id-token` when it expires.

This would also enable uses to use a [odic-login](https://github.com/int128/kubelogin) plugin for their `kubeconfig` in case their `refresh-token` is revoked.

The same approach can also be applied for Shoot clusters with `oidc` configured.

### PR DESCRIPTION
**Pull request by holgerkoser**
**What this PR does / why we need it**:
Allow users to download a kubeconfig file for accessing the garden cluster. The kubeconfig file will oidc-login kubectl plugin for cluster authentication.

**New configuration properties example [`values.yaml`]**
```yaml
...
apiServerUrl: https://api.garden.example.org  # required - URL of kube-apiserver 
apiServerCaData: LS0tLS1CRUdJTiBDR...         # optional - caData of kube-apiserver 
...
oidc:
  issuer: https://identity.garden.example.org # required - URL of OIDC provider
  ...
  public:
    clientId: kube-kubectl              # optional - Id of the public OIDC client. Defaults to `kube-kubectl`. 
    clientSecret: <publicClientSecret>  # required - Secret of the public OIDC client  
...
```
**Advanced options collapsed**
Advanced options collapsed

**Advanced options expaned**
<div>
Screen Shot 2020-04-07 at 13 09 34

**Which issue(s) this PR fixes**:
Fixes #417

**Special notes for your reviewer**:

**Release note**:

```improvement user
Users are able to download a `kubeconfig` file for accessing the garden cluster with `kubectl`
```

### CODE REVIEW

File: frontend/src/pages/Account.vue [hunk 1]
```diff
@@ -15,61 +15,212 @@ limitations under the License.
...
+                    <a href="https://kubernetes.io/docs/reference/access-authn-authz/authentication/#openid-connect-tokens" target="_blank" class="teal--text text--darken-2">
+                      OIDC
+                    </a>
+                    authentication via <tt>kubelogin</tt>.
+                    If not already done, please install <tt>kubelogin</tt>
+                    according to the
+                    <a link href="https://github.com/int128/kubelogin#setup" target="_blank" class="teal--text text--darken-2">
+                      setup instructions
+                    </a>.
+                    For more information about please refer to the <tt>kubelogin</tt> documentation.
```
Comment (petersutter) [inline frontend/src/pages/Account.vue:133]:
```suggestion
                    For more information please refer to the <tt>kubelogin</tt> documentation.
```

File: frontend/src/pages/Account.vue [hunk 2]
```diff
@@ -15,61 +15,212 @@ limitations under the License.
...
+                    <br/>
+                    Below you can configure and preview some the <tt>kubeconfig</tt> file before download.
```
Comment (petersutter) [inline frontend/src/pages/Account.vue:135]:
```suggestion
                    Below you can configure and preview the <tt>kubeconfig</tt> file before download.
```

File: frontend/src/pages/Account.vue [hunk 3]
```diff
@@ -15,61 +15,212 @@ limitations under the License.
...
+                  <span>{{ expansionPanel ? 'Collapse advanced options' : 'Show advanced options' }}</span>
+                </v-tooltip>
+              </v-list-tile-action>
+            </v-list-tile>
+            <v-expand-transition>
+              <v-card v-if="expansionPanel" flat class="mx-2">
+                <v-card-text class="pt-0">
+                  <div class="grey--text text--darken-2">
+                    The downloaded <tt>kubeconfig</tt> will initiate
+                    <a href="https://kubernetes.io/docs/reference/access-authn-authz/authentication/#openid-connect-tokens" target="_blank" class="teal--text text--darken-2">
```
Comment (petersutter) [inline frontend/src/pages/Account.vue:124]: Add the arrow icon for links pointing to external pages

File: frontend/src/pages/Account.vue (hunk 2 above)
Comment (holgerkoser) [reply frontend/src/pages/Account.vue:135]: ✔️

File: frontend/src/pages/Account.vue (hunk 3 above)
Comment (holgerkoser) [reply frontend/src/pages/Account.vue]: Added new component `external-link`

File: frontend/src/pages/Account.vue (hunk 1 above)
Comment (holgerkoser) [reply frontend/src/pages/Account.vue]: ✔️

File: charts/gardener-dashboard/values.yaml [hunk 4]
```diff
@@ -50,6 +56,14 @@ oidc:
   clientSecret: ~
   # redirectUri is the endpoint receiving the OTAC from the OpenID provider after successful login
   redirectUri: https://dashboard.ingress.example.org/auth/callback
+  # configuration for kubeconfig download required by kubelogin
```
Comment (holgerkoser) [inline charts/gardener-dashboard/values.yaml:59]: move to `.Values.oid.public`

**Comment by grolu** [comment]
Add example to Pull Request

File: frontend/src/pages/Administration.vue [hunk 5]
```diff
@@ -76,6 +77,14 @@ limitations under the License.
             <label class="caption grey--text text--darken-2">{{slaTitle}}</label>
             <p class="subheading" v-html="slaDescriptionCompiledMarkdown" />
           </v-flex>
+          <v-flex xs12>
```
Comment (holgerkoser) [inline frontend/src/pages/Administration.vue:80]: Only show if `publiv.clientSecret` is configured.

File: charts/gardener-dashboard/values.yaml (hunk 4 above)
Comment (holgerkoser) [reply charts/gardener-dashboard/values.yaml:59]: ✔️

**Comment by holgerkoser** [comment]
> Add example to Pull Request
✔️

File: frontend/src/pages/Administration.vue (hunk 5 above)
Comment (holgerkoser) [reply frontend/src/pages/Administration.vue]: ✔️

File: frontend/src/pages/Account.vue [hunk 6]
```diff
@@ -105,7 +276,109 @@ export default {
...
+        user: 'oidc-login'
+      }
+      if (namespace) {
+        context.namespace = namespace
+      }
+      const args = [
+        'oidc-login',
+        'get-token',
+        '--oidc-issuer-url=' + issuerUrl,
+        '--oidc-client-id=' + clientId,
```
Comment (petersutter) [inline frontend/src/pages/Account.vue:325]: Isn‘t the client-id optional?

Comment (petersutter) [inline frontend/src/pages/Account.vue:325]: Isn‘t the client-id optional? If it is not set, will it be `—oidc-client-id=undefined` ?

File: frontend/src/router/index.js [hunk 7]
```diff
@@ -509,6 +509,7 @@ export default function createRouter ({ store, userManager }) {
       await Promise.all([
         ensureCloudProfilesLoaded(),
         ensureProjectsLoaded(),
+        store.dispatch('fetchKubeconfigData'),
```
Comment (petersutter) [inline frontend/src/router/index.js:512]: Shouldn‘t we call it only for the accounts page?

Comment (holgerkoser) [reply frontend/src/router/index.js:512]: The method `fetchKubeconfigData` only makes a remote call if `state.kubeconfigData` is empty.

**Pull request merged**